### PR TITLE
Ui improvements

### PR DESCRIPTION
**Pull request by TontonBuck**
- Improved assets quality by adding 2x and 3x sizes. Changed some icons
- Added a template of selection view, edited materials, equipments, and persons views to used that template
- Reorganized and renamed outlets of AddInterventionVC
- Renamed extensions, removed equipment and person selection and creation views from storyboard because they are initialized programmatically
- Edited equipment and person selection, now using cell created programmatically instead of interface builder. "row" attributes no longer needed in core data
- Added error checks of creation views, preventing user to create objects with empty names or names already used

### CODE REVIEW

File: Controllers/AddInterventionViewController.swift [hunk 1]
```diff
@@ -13,80 +13,68 @@ class AddInterventionViewController: UIViewController, UITableViewDelegate, UITa
...
   @IBOutlet weak var materialsAddButton: UIButton!
   @IBOutlet weak var materialsCountLabel: UILabel!
-  @IBOutlet weak var materialsExpandImage: UIImageView!
+  @IBOutlet weak var materialsExpandImageView: UIImageView!
   @IBOutlet weak var selectedMaterialsTableView: UITableView!
   @IBOutlet weak var materialsTableViewHeightConstraint: NSLayoutConstraint!
+  @IBOutlet weak var materialsSeparatorView: UIView!
+
+  // Equipments
+  @IBOutlet var equipmentsTapGesture: UITapGestureRecognizer!
```
Comment (Jonathan-dehaay) [inline Controllers/AddInterventionViewController.swift]: Could it be a weak var like others outlet ?

File: Controllers/AddInterventionViewController.swift [hunk 2]
```diff
@@ -13,80 +13,68 @@ class AddInterventionViewController: UIViewController, UITableViewDelegate, UITa
...
+  @IBOutlet weak var addInputsButton: UIButton!
+  @IBOutlet weak var inputsCollapseButton: UIButton!
+  @IBOutlet weak var inputsNumber: UILabel!
+  @IBOutlet weak var selectedInputsTableView: UITableView!
+  @IBOutlet weak var selectedInputsTableViewHeightConstraint: NSLayoutConstraint!
+  @IBOutlet weak var inputsSeparatorView: UIView!
+
   // Materials
-  @IBOutlet weak var materialsHeightConstraint: NSLayoutConstraint!
   @IBOutlet var materialsTapGesture: UITapGestureRecognizer!
```
Comment (Jonathan-dehaay) [inline Controllers/AddInterventionViewController.swift]: Could it be a weak var like others outlet ?

File: Controllers/DetailedInterventionViewController+Irrigation.swift [hunk 3]
```diff
@@ -75,33 +74,32 @@ extension AddInterventionViewController: UITextFieldDelegate, CustomPickerViewPr
...
-    irrigationLabel.text = String(format: "Volume • %g %@", volume, unit)
+    selectedIrrigationLabel.text = String(format: "Volume • %g %@", volume, unit)
     updateInfoLabel(volume, unit)
   }
 
   private func updateInfoLabel(_ volume: Float, _ unit: String) {
     if volume == 0 {
-      irrigationInfoLabel.text = "Le volume ne peut être nul"
-      irrigationInfoLabel.textColor = AppColor.TextColors.Red
+      irrigationErrorLabel.text = "Le volume ne peut être nul"
```
Comment (Jonathan-dehaay) [inline Controllers/DetailedInterventionViewController+Irrigation.swift]: Should be localized strings.

File: Controllers/DetailedInterventionViewController+Irrigation.swift [hunk 4]
```diff
@@ -75,33 +74,32 @@ extension AddInterventionViewController: UITextFieldDelegate, CustomPickerViewPr
...
+      irrigationErrorLabel.textColor = AppColor.TextColors.Red
     } else if totalLabel.text == "+ SÉLECTIONNER" {
-      irrigationInfoLabel.text = "Aucune culture sélectionnée"
-      irrigationInfoLabel.textColor = AppColor.TextColors.Red
+      irrigationErrorLabel.text = "Aucune culture sélectionnée"
```
Comment (Jonathan-dehaay) [inline Controllers/DetailedInterventionViewController+Irrigation.swift]: Should be localized strings.

File: Controllers/DetailedInterventionViewController+Irrigation.swift [hunk 5]
```diff
@@ -75,33 +74,32 @@ extension AddInterventionViewController: UITextFieldDelegate, CustomPickerViewPr
...
+      irrigationErrorLabel.textColor = AppColor.TextColors.Red
     } else {
       let efficiency = volume / cropsView.selectedSurfaceArea
 
-      irrigationInfoLabel.text = String(format: "Soit %.1f %@ par hectare", efficiency, unit)
-      irrigationInfoLabel.textColor = AppColor.TextColors.DarkGray
+      irrigationErrorLabel.text = String(format: "Soit %.1f %@ par hectare", efficiency, unit)
```
Comment (Jonathan-dehaay) [inline Controllers/DetailedInterventionViewController+Irrigation.swift]: Should be localized strings.

File: Controllers/DetailedInterventionViewController+Irrigation.swift [hunk 6]
```diff
@@ -75,33 +74,32 @@ extension AddInterventionViewController: UITextFieldDelegate, CustomPickerViewPr
...
   }
 
   @objc public func updateIrrigation(_ sender: Any) {
-    let volumeString = irrigationValueTextField.text!.replacingOccurrences(of: ",", with: ".")
-    let volume = Float(volumeString) ?? 0
+    let volume = irrigationVolumeTextField.text!.floatValue
     let unit = irrigationUnitButton.titleLabel!.text!
 
-    irrigationLabel.text = String(format: "Volume • %g %@", volume, unit)
+    selectedIrrigationLabel.text = String(format: "Volume • %g %@", volume, unit)
```
Comment (Jonathan-dehaay) [inline Controllers/DetailedInterventionViewController+Irrigation.swift]: Should be localized strings.

File: Controllers/DetailedInterventionViewController+Irrigation.swift [hunk 7]
```diff
@@ -56,11 +56,10 @@ extension AddInterventionViewController: UITextFieldDelegate, CustomPickerViewPr
       dimView.isHidden = true
       harvestTableView.reloadData()
     case irrigationPickerView:
-      let volumeString = irrigationValueTextField.text!.replacingOccurrences(of: ",", with: ".")
-      let volume = Float(volumeString) ?? 0
+      let volume = irrigationVolumeTextField.text!.floatValue
 
       self.irrigationUnitButton.setTitle(selectedValue, for: .normal)
-      irrigationLabel.text = String(format: "Volume • %g %@", volume, unit)
+      selectedIrrigationLabel.text = String(format: "Volume • %g %@", volume, unit)
```
Comment (Jonathan-dehaay) [inline Controllers/DetailedInterventionViewController+Irrigation.swift]: Should be localized strings.

**Comment by Aquaj** [comment]
Approved once strings have been localized.

File: Controllers/AddInterventionViewController.swift [hunk 8]
```diff
@@ -13,80 +13,68 @@ class AddInterventionViewController: UIViewController, UITableViewDelegate, UITa
...
 
-  // Equipment
-  @IBOutlet weak var equipmentDarkLayer: UIView!
-  @IBOutlet weak var equipmentName: UITextField!
-  @IBOutlet weak var equipmentNumber: UITextField!
-  @IBOutlet weak var equipmentType: UILabel!
-  @IBOutlet weak var equipmentHeightConstraint: NSLayoutConstraint!
-  @IBOutlet weak var equipmentTableViewHeightConstraint: NSLayoutConstraint!
+  // Persons
+  @IBOutlet var personsTapGesture: UITapGestureRecognizer!
```
[the remark this answers is not in the document]
Comment (TontonBuck) [reply Controllers/AddInterventionViewController.swift]: No, Xcode doesn't make it weak when you create a gesture outlet

**Pull request merged**